test/integration/card_details_errors_ft_tests.js

```diff
@@ -146,6 +146,12 @@ describe('chargeTests', function () {
   })
 
   describe('Enter card details page — errors ', function () {
+    it('should error without frontend state cookie', function (done) {
```

Comment:
Is it possible to have an integration test that visits the 3ds in endpoint specifically, once with a valid state cookie and once without a valid state cookie? That would ensure all the cookie + session logic is working at a high level.

I believe these routes are already covered in `/test/integration/three_d_secure_ft_tests.js` so it may be possible to just add assertions to that suite.